ztools/ztls/handshake_messages.go

```diff
@@ -762,6 +807,30 @@ func (m *serverHelloMsg) unmarshal(data []byte) bool {
 				return false
 			}
 			m.extendedMasterSecret = true
+
+		case extensionSCT:
+			d := data[:length]
+			if len(d) < 2 {
+				return false
+			}
+			l := int(d[0])<<8 | int(d[1])
+			d = d[2:]
+			if len(d) != l || l == 0 {
+				return false
+			}
+			m.scts = make([][]byte, 0, 3)
+			for len(d) != 0 {
+				if len(d) < 2 {
+					return false
+				}
+				sctLen := int(d[0])<<8 | int(d[1])
```

Comment:
You treat `sctlen` different in `serverHello.unmarshal` vs `serverHello.marshal`. Why is there only a loop in one of these functions?